httpx/client.py

```diff
@@ -896,3 +945,10 @@ def __exit__(
         traceback: TracebackType = None,
     ) -> None:
         self.close()
+
+
+def proxy_from_url(url: URLTypes) -> AsyncDispatcher:
+    url = URL(url)
+    if url.scheme in ("http", "https"):
+        return HTTPProxy(proxy_url=url)
+    raise ValueError(f"Unknown proxy URL: {url!r}")
```

Comment:
Should we raise a custom exception type here? Perhaps `InvalidURL`?

Also "Unknown" isn't very informative as to *why* the URL is unknown — something about the condition two lines about would be more helpful if we encounter this error as a user. :-)

Reply:
Maybe we should raise ProxyError? Yeah that message is not good, will fix.

Reply:
I don't know about `ProxyError` — right now it's documented as an error raised from *within* a proxy. I do think it's a case for an `InvalidURL` error here.

Reply:
Agreed, I'll have to add that.